R/weight.R

```diff
@@ -68,6 +89,49 @@ setMethod("weightVariables", "VariableCatalog", function (x) {
     }
 })
 
+#' @rdname weightVariables
+#' @export
+`weightVariables<-` <- function(x, value) {
+    if (is.dataset(x)) {
```

Comment:
This if... else if tells me that this is perhaps better done as an S4 method.

Reply:
I agree, it also would be easy to define both using the same code path with this pattern:
```
setMethod("weightVariables<-", "VariableCatalog", function (x, value) {
    #work
})

setMethod("weightVariables<-", "CrunchDataset", function (x, value) {
    varCat <- allVariables(x)
    weightVariables(varCat) <- value
})
```